source/neuropod/bindings/java/src/test/java/com/uber/neuropod/NeuropodTest.java

```diff
@@ -122,6 +132,135 @@ public void loadModel() {
+
+       assertArrayEquals(new long[]{1L, 2L}, tensor.getDims());
+       assertEquals(2, tensor.getNumberOfElements());
+       assertEquals(TensorType.INT32_TENSOR, tensor.getTensorType());
+
+       tensor.close();
+       allocator.close();
+    }
+
+    @Test
+    public void InputLongTensor() {
+       NeuropodTensorAllocator allocator = model.getTensorAllocator();
+
+       TensorType type = TensorType.INT64_TENSOR;
+       ByteBuffer buffer = ByteBuffer.allocateDirect(type.getBytesPerElement() * 2).order(ByteOrder.nativeOrder());
+
+       LongBuffer typedBuffer = buffer.asLongBuffer();
+       typedBuffer.put(1);
+       typedBuffer.put(2);
+       NeuropodTensor tensor = allocator.tensorFromMemory(buffer, new long[]{1L, 2L}, type);
+       assertNotNull(tensor.toLongBuffer());
+
+       assertArrayEquals(new long[]{1L, 2L}, tensor.getDims());
+       assertEquals(2, tensor.getNumberOfElements());
+       assertEquals(TensorType.INT64_TENSOR, tensor.getTensorType());
+
+       tensor.close();
+       allocator.close();
+    }
+
+    @Test
+    public void infer() {
+        NeuropodTensorAllocator allocator = model.getTensorAllocator();
+        Map<String, NeuropodTensor> inputs = new HashMap<>();
+        TensorType type = TensorType.FLOAT_TENSOR;
+
+        ByteBuffer bufferX = ByteBuffer.allocateDirect(type.getBytesPerElement() * 2).order(ByteOrder.nativeOrder());
+        FloatBuffer floatBufferX = bufferX.asFloatBuffer();
+        floatBufferX.put(1.0f);
+        floatBufferX.put(2.0f);
+        NeuropodTensor tensorX = allocator.tensorFromMemory(bufferX, new long[]{1L, 2L}, type);
+        inputs.put("x", tensorX);
+
+        ByteBuffer bufferY = ByteBuffer.allocateDirect(type.getBytesPerElement() * 2).order(ByteOrder.nativeOrder());
+        FloatBuffer floatBufferY = bufferY.asFloatBuffer();
+        floatBufferY.put(3.0f);
+        floatBufferY.put(4.0f);
+        NeuropodTensor tensorY = allocator.tensorFromMemory(bufferY, new long[]{1L, 2L}, type);
+        inputs.put("y", tensorY);
+
+        Map<String, NeuropodTensor> res = model.infer(inputs);
+        assertEquals(1, res.size());
+
+        assertTrue(res.containsKey("out"));
+        NeuropodTensor out = res.get("out");
+        assertNotNull(out);
+        FloatBuffer outBuffer = out.toFloatBuffer();
+
+        assertArrayEquals(new long[]{1L, 2L}, out.getDims());
+        assertEquals(2, out.getNumberOfElements());
+        assertEquals(TensorType.FLOAT_TENSOR, out.getTensorType());
+
+        assertEquals(4.0f , outBuffer.get(0), EPSILON);
+        assertEquals(6.0f , outBuffer.get(1), EPSILON);
+
+        out.close();
+
+        // Inference with requested outputs.
+        List<String> requestedOutputs = new ArrayList<String>();
+        requestedOutputs.add("out");
+        Map<String, NeuropodTensor> res2 = model.infer(inputs, requestedOutputs);
+        assertEquals(1, res2.size());
+
+        try
+        {
+           // Try failuer if provide invalid output.
```

Comment:
nit: typo

Reply:
"Add unexpected "requested output" that should cause a failure."